azure-functions-sdk/src/commands/new.rs

```diff
@@ -0,0 +1,461 @@
+                Arg::with_name("name")
+                    .long("name")
+                    .short("n")
+                    .value_name("NAME")
+                    .help("The name of the new Azure Function.")
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name("queue_name")
+                    .long("queue-name")
+                    .short("q")
+                    .value_name("QUEUE")
+                    .help("The name of the storage queue to monitor.")
+                    .required(true),
+            )
+    }
+
+    pub fn execute(&self, quiet: bool) -> Result<(), String> {
+        Queue::validate_queue_name(self.queue_name)?;
+
+        create_function(
+            self.name,
+            &templates!(
+            "new" =>
+            [
+                "queue.rs"
+            ]),
+            "queue.rs",
+            &json!({
+                "name": self.name,
+                "queue_name": self.queue_name
+            }),
+            quiet,
+        )
+    }
+
+    fn validate_queue_name(name: &str) -> Result<(), String> {
+        if name.len() < 3 {
+            return Err(format!(
+                "queue name '{}' must be at least 3 characters.",
+                name
+            ));
+        }
+
+        if name.len() > 63 {
+            return Err(format!(
+                "queue name '{}' cannot be more than 63 characters.",
+                name
+            ));
+        }
+
+        if !Regex::new("^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]$")
+            .unwrap()
+            .is_match(name)
+        {
+            return Err(format!("queue name '{}' must start and end with a letter or number and can only contain letters, numbers, and hyphens.", name));
+        }
+
+        Ok(())
+    }
+}
+
+impl<'a> From<&'a ArgMatches<'a>> for Queue<'a> {
+    fn from(args: &'a ArgMatches<'a>) -> Self {
+        Queue {
+            name: args.value_of("name").unwrap(),
+            queue_name: args.value_of("queue_name").unwrap(),
+        }
+    }
+}
+
+struct Timer<'a> {
+    name: &'a str,
+    schedule: &'a str,
+}
+
+impl<'a> Timer<'a> {
+    pub fn create_subcommand<'b>() -> App<'a, 'b> {
+        SubCommand::with_name("timer")
+            .about("Creates a new trimer triggered Azure Function.")
+            .arg(
+                Arg::with_name("name")
+                    .long("name")
+                    .short("n")
+                    .value_name("NAME")
+                    .help("The name of the new Azure Function.")
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name("schedule")
+                    .long("schedule")
+                    .short("s")
+                    .value_name("SCHEDULE")
+                    .help("The timer schedule as a cron-expression.")
+                    .required(true),
+            )
+    }
+
+    pub fn execute(&self, quiet: bool) -> Result<(), String> {
+        create_function(
```

Comment:
nit: I find the various calls to `create_function` hard to read. Maybe puting some things in temp variables above the function call could help readability?

Reply:
I'd be happy to.  I'll push that up shortly.